benches/hook_performance_impact.rs

```diff
@@ -52,14 +52,16 @@ fn criterion_benchmark(c: &mut Criterion) {
     }
 
     let mut instance_non_empty_hookset =
-        RuntimeInstance::new_with_hooks(&validation_info, MyCustomHookSet).unwrap();
+        RuntimeInstance::new_with_hooks(DEFAULT_MODULE, &validation_info, MyCustomHookSet).unwrap();
 
     c.bench_function("invoke_func EmptyHookSet", |b| {
-        b.iter(|| instance_empty_hookset.invoke_func::<_, ()>(black_box(2), black_box(42_i32)))
+        let test_fn = instance_empty_hookset.get_fn_idx(0, 2).unwrap();
```

Comment:
I think it is more common to do the setup for benchmarks outside of `bench_function`.
Criterion.rs has only example that kinda explains this for benchmarks with inputs:
```rs
// Setup (construct data, allocate memory, etc)
let input = 5u64;
c.bench_with_input(
    BenchmarkId::new("function_name", input), &input,
    |b, i| b.iter(|| {
        // Code to benchmark using input `i` goes here
    }),
);
```
https://docs.rs/criterion/latest/criterion/struct.Criterion.html#example-1

Reply:
I put the `test_fn` initialization outside the `bench_function`. I believe `bench_with_input` is more appropriate for something with an input that can vary? I'm not sure, though.